src/rascal/representations/calculator_spherical_expansion.hh

```diff
@@ -2181,6 +2191,67 @@ namespace rascal {
       expansions_coefficients_gradient.set_shape(ThreeD * n_row, n_col);
     }
 
+    // species in projection matrices have to be checked here, because
+    // calculators do not have access to manager during construction
+    if (hypers.count("radial_contribution") &&
+        hypers.at("radial_contribution").get<json>().count("optimization") &&
+        hypers.at("radial_contribution")
+            .get<json>()
+            .at("optimization")
+            .get<json>()
+            .count("RadialDimReduction")) {
```

Comment:
in set_hyperparameters() you already know if there is a RadialDimReduction so I would set a flag there to simplify this block.

Reply:
It is since `compute_neighbour_contribution` is called `n_neigh` times instead of `n_atom` times. In any case I still feel that it's cleared to have this check in `compute_impl` with a loop over atoms and a flag to check if `RadialDimReduction` is on but I don't want to argue about it either. It works well as is, you could change `.at()` with a direct access since at that point you know that you can access this element.

Reply:
Yes but for a correct check one needs to check all neighbours types, since one could mask central species atoms, which then only appear as neighbours and not anymore as central atoms I had this problem with the chemical shielding dataset.
So I need to check all central atom types and all neighbour types

Reply:
I did not think of that right.